getting_started/step_by_step/resources.rst

```diff
@@ -158,12 +158,173 @@ Freeing resources
+
+- Resources can even serialize sub-Resources recursively, meaning users can design even more sophisticated data structures.
+
+- Users can save Resources as version-control-friendly text files (\*.tres). Upon exporting a game, Godot serializes resource files as binary files (\*.res) for increased compression.
+
+- Godot Engine's Inspector renders and edits Resource files out-of-the-box. As such, users do not need to implement custom logic to visualize or edit their data. To do so, double-click the resource file in the FileSystem dock or click the folder icon in the Inspector and open the file in the dialog.
+
+- They can extend **other** resource types besides just the base Resource.
+
+Godot makes it easy to create custom Resources in the Inspector.
+
+1. Create a plain Resource object in the Inspector. This can even be a type that derives Resource, so long as your script is extending that type.
+2. Set the ``script`` property in the Inspector to be your script.
+
+The Inspector will now display your Resource script's custom properties. If one edits
+those values and saves the resource, the Inspector serializes the custom properties
+too! To save a resource from the Inspector, click the Inspector's tools menu (top right),
+and select "Save" or "Save As...".
+
+If the script's language supports :ref:`script classes <register-scripts-as-classes>`, 
+then it streamlines the process. Defining a name for your script alone will add it to
+the Inspector's creation dialog. This will auto-add your script to the Resource
+object you create.
+
+Let's see some examples.
+
+.. tab::
+  ..code-tab:: gdscript GDScript
+    # bot_stats.gd
+    extends Resource
+    export(int) var health
+    export(Resource) var sub_resource
+    export(Array, String) var str_array
+
+    func _init(p_health = 0, p_sub_resource = null, p_str_array = []):
+        health = p_health
+        sub_resource = p_sub_resource
+        str_array = p_str_array
+
+    # bot.gd
+    extends KinematicBody
+    export(Resource) var stats
+
+    const BotStats = preload("res://bot_stats.gd")
+
+    func _init():
+        if not stats:
+            stats = BotStats.new(10)
+    
+    func _ready():
+        print(stats.health) # prints '10'
+  ..code-tab:: csharp
+        using Godot;
+
+        public class Bot : KinematicBody
+        {
+            [Export]
+            public BotStats stats;
+
+            public override void _Init()
+            {
+                if (!stats)
+                {
+                    stats = (ResourceLoader::load("BotStats.cs") as BotStats).new(10);
+                }
+            }
+
+            public override void _Ready()
+            {
+                Godot.print(stats.health);
+            }
+        }
+
+.. note::
+
+   Resource scripts are similar to Unity's ScriptableObjects. However, unlike in
+   Unity, there is no need to implement IMGUI logic to enable users to create
```

Comment:
In fact, there is no need to implement immediate gui or custom editor to use a "common" ScriptableObject.

The current workflow (I tested it in Unity 2018.3):

1. Create your `ScriptableObject` class:
```c#
public class CustomData : ScriptableObject {
    public string data;
}
```
2. Add it to `Assets->Create` menu:
```c#
// This is equivalent to `class_name` in Godot 3.1+
[CreateAssetMenu(menuName="MyCustomStuff/CustomData")]
public class CustomData : ScriptableObject {
    public string data;
}
```
3. Create a component that will use your `ScriptableObject`:
```c#
public class MyComponent : MonoBehaviour {
    public CustomData data;
}
```
4. Create your asset using the `Assets->Create` menu;
5. Drag and drop it to the `data` field in the inspector.

Reply:
Oh, interesting. That's news to me. Great that they added that feature! Appreciate the heads-up. I'll go ahead and take out that tidbit then.

Reply:
Unity docs and changelogs are pretty confusing so I don't know in which version this started to work this way :sweat_smile: